### DISCUSSION TITLE
Nodejs e php

### DISCUSSION THREAD

**Discussion by wellpinho** [frontendbr/forum#1953] [Perguntas e Respostas]
_Dev boa tarde.

Da pra usar NodeJS e hospedar sites simples com php no mesmo dedicado da Digital Ocean? 
Ou isso gera conflito de ip entre node e o apache?_

**Comment by portothree** [comment]
Ola @wellpinho. 

Sim, é possível hospedar inúmeros _sites_ no mesmo servidor, não haverá conflito de IP, mas talvez de porta.
Se subir o seu _PHP_ com _Apache_ na porta 80 e o seu servidor _Node_ também na porta 80, terá sim, problemas. Entretanto, não é recomendado que deixe o seu servidor _Node_ exposto diretamente na porta 80, melhor seria também utilizar o Apache, mas criar diferentes pastas em `/var/www` e configurações em `/etc/apache2/sites-available/`.
Exemplo:
1. `/var/www/domain.com/public_html` e `cp /etc/apache2/sites-available/000-default.conf /etc/apache2/sites-available/domain.com.conf`
2. `/var/www/domain2.com/public_html` e `cp /etc/apache2/sites-available/000-default.conf /etc/apache2/sites-available/domain2.com.conf`


Talvez https://github.com/backend-br/forum seja o local mais apropriado para essa dúvida, mas espero ter ajudado. Abraço,

**Comment by wilsonneto-swe** [comment]
Bom dia @wellpinho , complementando a resposta do @portothree, você consegue sim, sendo dedicado é como a sua própria máquina, você consegue rodar node, php, python e etc se quiser, o problema mesmo será a porta, já que a default web é a 80. 

Neste cenário você precisa de um proxy-reverso, um nginx talvez, o proxy reverso escutará na porta 80 e se ele vê que a requisição é para o app node ele direciona para o app em node, se ele ver que a requisição é para o app em PHP ele direciona para ele. 

Penso que uma boa ideia neste cenário é trabalhar com containers, um container sendo o app em node, outro sendo o app em php e outro sendo o proxy reverso que irá escutar a 80/443, assim você precisará preparar o ambiente de sua droplet apenas para trabalhar com docker ou outra engine de container, não vai precisar preparar ambiente para php, para node, instalar nginx ou outro...

Espero que ajude,
forte abraço!

**Reply by portothree** [reply]
Apache também faz proxy reverso, mas também prefiro nginx e a sua sugestão de containers parece-me uma boa.
